src/OpenLoco/Paint/Paint.h

```diff
@@ -331,6 +324,22 @@ namespace OpenLoco::Paint
         //Map::TileElement* trackElementOnSameHeight;
         //uint8_t unk141E9DB;
         //uint32_t trackColours[4];
+        template<typename T>
+        T* allocatePaintStruct()
+        {
+            auto* ps = *_nextFreePaintStruct;
+            if (ps >= *_endOfPaintStructArray)
+            {
+                return nullptr;
+            }
+            *_nextFreePaintStruct = reinterpret_cast<PaintEntry*>(reinterpret_cast<uintptr_t>(*_nextFreePaintStruct) + sizeof(T));
```

Comment:
I think you can just do this:
`*_nextFreePaintStruct += sizeof(T);`
there is no need to cast back and forth between pointer types

Reply:
Pointer arithmetic increments by type not by bytes.

Reply:
Dang it I had a strange feeling wrote that comment but I couldn't figure out what it was; I'd totally forgotten c++ had this, it's been a long time since I've needed this knowledge :( thanks for the info!